src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -412,6 +418,14 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		}
 
 		let newProfile = new ConnectionProfile(this._capabilitiesService, model || providerName);
+		if (!model.password) {
```

Comment:
this should only be performed for sql login auth type, right?

Reply:
It isn't able to retrieve anything if the auth type isn't SQL. login.

Reply:
The authentication type is also being checked on line 413 - 414, so it seemed redundant to check here too.

Reply:
fine by me.

Reply:
What checks are you talking about? I only see it getting the default value and using that if the model doesn't have one itself. But that doesn't affect this line so it'll try to fill in the password regardless of the auth type.

Reply:
The checks in the `addSavedPassword` function. If the auth type is SqlLogin then it will set `isPasswordRequired` to `true`, and go to the credential service to read the password.
![image](https://user-images.githubusercontent.com/87730006/218878478-3c8c7f22-19a6-44c2-967d-f607f8b5218f.png)

Reply:
Ok good, was confused with the line numbers you gave earlier. That seems fine then.